### ISSUE TITLE
Request translations for pdf_downloader app

### ISSUE THREAD

**Issue by rotdrop** [nextcloud/docker-ci#467]
# 👤 App repository

**To be done by you**

- [x] Link to your repository: https://github.com/rotdrop/nextcloud-app-pdf-downloader
- [x] If the app is personal repository
    - [x] Invite `nextcloud-bot` with write permissions
    - [x] Make sure all translated branches do **not** have branch protection enabled, see https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#setup-of-the-transifex-sync for more details
- [x] Create file `l10n/.gitkeep` with empty content
- [x] Add `.l10nignore` to exclude compiled JS files and thirdparty code, e.g. ignoring compiled javascript assets and composer PHP dependencies in the vendor/ directory:
```
js/
vendor/
```
- [x] Create file `.tx/config` with the following content, replace `{{APPID}}` with your app id 3 times:
```ini
[main]
host     = https://www.transifex.com
lang_map = hu_HU: hu, nb_NO: nb, sk_SK: sk, th_TH: th, ja_JP: ja, bg_BG: bg, cs_CZ: cs, fi_FI: fi

[o:nextcloud:p:nextcloud:r:{{APPID}}]
file_filter = translationfiles/<lang>/{{APPID}}.po
source_file = translationfiles/templates/{{APPID}}.pot
source_lang = en
type        = PO
```

# 🏗️ Sysadmin team
- [x] 👀 Ensure access:
    - [ ] Nextcloud-Org repository: Add `nextcloud-bot` with admin permissions
    - [x] Other repositories:
        - [x] Ensure the invite was accepted
        - [x] Ensure the app certificate is linked to the same repository/author https://github.com/nextcloud/app-certificate-requests
- [x] ⚙️ Ensure repository setup:
    - [x] `.tx/config`
    - [x] `l10n/.gitkeep`
    - [x] `.l10ignore`
- [x] ➕ Add `"Owner Repository",` into https://github.com/nextcloud/docker-ci/edit/master/translations/config.json
    - [x] Pull request: https://github.com/nextcloud/docker-ci/pull/470
- [x] 🔑 SSH into translation machine: `ssh root@transifex-sync.nextcloud.com`
- [x] ↩️ Change dir: `cd /srv/docker-ci`
- [x] ⬇️ Pull: `git pull origin master`
- [x] 🧑‍💻 Run the first sync manually:

```sh
docker run -v /srv/cronie-data/transifexrc:/root/.transifexrc \
    -v /srv/cronie-data/gpg:/gpg \
    -v /srv/cronie-data/ssh/id_rsa:/root/.ssh/id_rsa \
    --rm -i ghcr.io/nextcloud/continuous-integration-translations-app \
    AUTHOR APPID
```

# 🔣 Translation team
- [x] Transifex: New resource should appear
- [x] Transifex: Do translations and check sync to repo
- [ ] Add resource to [wiki](https://help.nextcloud.com/t/list-of-resources-and-their-priority-for-translation/78312/)

**Comment by nickvergessen** [comment]
> I suppose that in addition to the checklist above I have also to remove any existing translations that have been done by myself?

The script of our translation sync does that automatically. But also okay to be done manually of course.

**Comment by nickvergessen** [comment]
Since you did not tick it:

> If the app is personal repository
> Invite nextcloud-bot with write permissions

Did you invite the bot already? That is necessary because that account will push the translations back to your repository

**Comment by rotdrop** [comment]
Ah, sorry. Misunderstood this. The project is public but of course, not everyone has write permissions. I suppose "the repository is private" means that it does not belong to an organisation. Should have been clear to me ...

**Comment by nickvergessen** [comment]
> means that it does not belong to an organisation

In fact it's even "Not to our Nextcloud organisation"

Let me rephrase the template

**Comment by rotdrop** [comment]
Perhaps one further comment: the template states "invite with write permissions". I think that I can only invite people, which hopefully will then have write-permissions (as the repo is already readable). But there are no controls to set the permission level.

**Comment by nickvergessen** [comment]
Yeah should be good enough.

**Comment by rotdrop** [comment]
I meant: perhaps the instructions could be adjusted? I was searching for the write-permission controls in the Github UI. But maybe I am the only one with this experience, so ... just leave it.

**Comment by rotdrop** [comment]
> > I suppose that in addition to the checklist above I have also to remove any existing translations that have been done by myself?
> 
> The script of our translation sync does that automatically. But also okay to be done manually of course.

- How to handle the case when there are issues with the translations received from transifex?
- Is it possible to upload existing translations first to transifex in order to speed up the initial translation process?

**Comment by p-bo** [comment]
@rotdrop
 > * How to handle the case when there are issues with the translations received from transifex?

Do you mean that something is wrongly translated?

 > * Is it possible to upload existing translations first to transifex in order to speed up the initial translation process?

There is upload file button on Transifex, not sure regarding import data format there
![obrazek](https://user-images.githubusercontent.com/12208593/201335350-36c5bf1c-bdf1-42c6-9d83-3e3cddbbb401.png)

**Comment by rotdrop** [comment]
> @rotdrop
> 
> > * How to handle the case when there are issues with the translations received from transifex?
> 
> Do you mean that something is wrongly translated?

Yes.

> 
> > * Is it possible to upload existing translations first to transifex in order to speed up the initial translation process?
> 
> For each resource (=app) and each language for it, there is upload file button on Transifex, not sure regarding import data 

Thanks for your hints! I will have look once things are set up for the app.

**Comment by p-bo** [comment]
> > @rotdrop
> > > * How to handle the case when there are issues with the translations received from transifex?
> > 
> > 
> > Do you mean that something is wrongly translated?
> 
> Yes.
Then fix it on Transifex (after few hours it will be synchronized into source code on git) if you are member of translation team of that language. If it is for other language, then there is still possible to insert better translation and save it as suggestion (and perhaps write - trough messages on Transifex - about it to previous translator / lang admin to use it instead).
> 
> > > * Is it possible to upload existing translations first to transifex in order to speed up the initial translation process?
> > 
> > 
> > For each resource (=app) and each language for it, there is upload file button on Transifex, not sure regarding import data
> 
> Thanks for your hints! I will have look once things are set up for the app.
Glad to help :-)

**Comment by nickvergessen** [comment]
Sync worked:
https://github.com/rotdrop/nextcloud-app-pdf-downloader/commit/4e5701b84903f36cd309e7352ac066ddf92bfa22
